src/llmcompressor/observers/hist.py

```
@@ -0,0 +1,48 @@
+
+@Observer.register("histogram")
+class HistogramObserver(Observer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        observer_kwargs = self.args.observer_kwargs
+        self.bins = observer_kwargs.get("bins", 2048)
+        self.histogram = None
+        self.xmax = -torch.inf
+
+    @torch.no_grad
+    def forward(self, observed: torch.Tensor) -> ScaleZpTuple:
```

Ruff (0.15.11)

[error] Undefined name `ScaleZpTuple`

(F821)